### PR TITLE
remove content_origin property from layout

### PR DESCRIPTION
**Pull request by Frizi**
### Pull Request Description

Simplified layout algorithm by removing `content_origin`, and instead treating `(0.0, 0.0)` as origin point in every layout object. This change allows overflowing containers that are within auto-layout. The parent element will no longer be moved within the grid cell when its children overflow it.

image

### Important Notes

When implementing this change, I have found that when object's size was modified without ever touching its position, that change was not being picked up in the "modified children" list, and `on_updated` was never triggered. Because some sprites now are bottom-left aligned, that is now a common case and was reproducible on the auto-layout example scene. I ended up fixing it by introducing another dirty flag for `computed_size` changes. Right now that flag is applied very broadly (on each layout update), but in the future we might make it more precise by actually checking if the size was changed in the process.

I believe that this might also be a fix for #5095, as I cannot reproduce it anymore with those changes.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by wdanilo** [comment]
` but in the future we might make it more precise by actually checking if the size was changed in the process.` - please implement it right now. It looks like a super simple addition and it will probably remove a LOT of unnecessary computations, ok?

File: lib/rust/ensogl/core/src/display/object/instance.rs [hunk 1]
```diff
@@ -939,80 +939,6 @@
...
-//! ```
-//! //     ╭──────── ▶ ◀ ──────────────────╮
-//! //     │ root                          │
-//! //     │                 ╭────────╮    │
-//! // ╭───┼────╮            │ node3  │    │
-//! // │ node1  │            │        │    ▼
-//! // │   │    │ ╭────────╮ ╰────────╯    ▲
-//! // ╰───┼────╯ │ node2  │               │
-//! //     ╰──────┼────────┼───────────────╯
-//! //   ╱        │        │
```
Comment (wdanilo) [inline lib/rust/ensogl/core/src/display/object/instance.rs]: 1. Docs should mention that the auto layout works only with bottom-left aligned things. This should be a big warning with a proper explanation of why old shapes will not work correctly. Also, I think that now, instead of having alignment as parameter to `ensogl::shape!` we should rename it to `ensogl::shape_old2` or something like that and keep `ensogl::shape!` always aligned to bottom left corner.
2. Could you please update docs with explicit info that the origin is always bottom left corner, please?
3. Could you add to docs example with an overflow of manual layout, please?

File: lib/rust/ensogl/core/src/display/object/instance.rs [hunk 2]
```diff
@@ -1814,6 +1750,11 @@ impl Model {
                 }
             } else {
                 trace!("Self origin and layers did not change.");
+
+                if self.dirty.computed_size.check() {
+                    self.on_updated_source.emit(());
```
Comment (wdanilo) [inline lib/rust/ensogl/core/src/display/object/instance.rs:1904]: missing trace

Comment (wdanilo) [inline lib/rust/ensogl/core/src/display/object/instance.rs:1904]: I don't understand the name `on_updated_source` - what does it mean? I see no docs next to this field. I know its not part of this PR, sorry for catching it late.

**Review by wdanilo — APPROVED** [review]

File: lib/rust/ensogl/core/src/display/object/instance.rs (hunk 2 above)
Comment (farmaazon) [reply lib/rust/ensogl/core/src/display/object/instance.rs:1904]: Looks like "A Source (i.e. frp::Source) for `on_updated`" event.

File: lib/rust/ensogl/core/src/display/object/instance.rs (hunk 1 above)
Comment (Frizi) [reply lib/rust/ensogl/core/src/display/object/instance.rs]: I would like to avoid introducing another deprecated shape macro. Instead, I will change the default to bottom_left, and override it in all existing shapes to center.

File: lib/rust/ensogl/core/src/display/object/instance.rs [hunk 3]
```diff
@@ -939,6 +939,117 @@
...
+//! //    ║ │╭ node1 ─ ▶ ◀ ──────╮┆╭ node2 ─ ▶ ◀ ──────╮│ ║
+//! //   ╭╫─┼┼─────╮           ╭─┼┼┼─────╮             ││ ║
+//! //   │║node1_1 │           │ node2_1 │             ││ ▼
+//! //   │║ ││     │           │ │┆│     │             ││ ▲
+//! //   │║ ││     │  ╭────────┼╮│┆│     │  ╭─────────╮││ ║
+//! //   ╰╫─┼┼─────╯  │ node1_2╰┼┼┼┼─────╯  │ node2_2 │││ ║
+//! //    ║ │╰────────┼─────────┼╯┆╰────────┼─────────┼╯│ ║
+//! //    ║ ╰─────────┼─────────┼─┴─────────┼─────────┼─╯ ║
+//! //    ╚═══════════╰─────────╯═══════════╰─────────╯═══╝
+//!
```
Comment (vitvakatu) [inline lib/rust/ensogl/core/src/display/object/instance.rs]: I'm impressed. How do you guys draw such things?

Comment (wdanilo) [reply lib/rust/ensogl/core/src/display/object/instance.rs]: by hand :P

File: lib/rust/ensogl/core/src/display/object/instance.rs (hunk 2 above)
Comment (wdanilo) [reply lib/rust/ensogl/core/src/display/object/instance.rs:1904]: oh, right!

**Review by wdanilo — APPROVED** [review]
